pkg/controllers/terminating/v1alpha1/terminate.go

```diff
@@ -38,110 +37,72 @@ type Terminator struct {
 	coreV1Client  corev1.CoreV1Interface
 }
 
-// cordonNodes takes in a list of expired nodes as input and cordons them
-func (t *Terminator) cordonNodes(ctx context.Context) error {
-	// 1. Get terminable nodes
-	nodeList, err := t.getLabeledNodes(ctx, provisioning.ProvisionerTerminablePhase)
-	if err != nil {
-		return err
+// cordonNode takes in a list of expired nodes as input and cordons them
+func (t *Terminator) cordonNode(ctx context.Context, node *v1.Node) error {
+	if node.Spec.Unschedulable {
```

Comment:
Do we need this check? The idempotency should be handled by the kubectl patch logic (if its the same, it won't make the call.)

Reply:
The Patch still makes a call here. When we make a patch, we need to make sure that the resource did not change while we were doing this logic. If the resource did change while we were doing this function, we can get a 409.

Reply:
Chatted offline. Removing this as the optimization here is pretty small if even present.